docs/recipes/recipe-data-structures-in-git/index.md

```diff
@@ -0,0 +1,366 @@
+---
+title: "Managing Data Structures in Git"
+description: "Learn how to utilize snowplow-cli and github actions to manage authoring and publishing for your data structures"
+sidebar_position: 41
+---
+
+# Introduction
+
+Snowplow Data Structures are the artifacts defining the rules for event validation within the Snowplow data pipeline. As such, they are a description of event shapes that the pipeline will allow through, and essentially the basis for the shape of data in the warehouse.
+
+The fact that Data Structures formalize what the warehouse tables look like makes them a cornerstone of Snowplow's facilities for data governance. An unintended change in a data structure could result in data consumers down the line being unable to process that data (e.g. data models breaking). That is why larger organizations guard Data Structure definitions closely, and need approval workflows to allow or disallow changes. For instance, a Data Protection Officer may want to have the final say about collected events, to ensure no PII is harvested.
+
+On top of that, detailed change history can be crucial for such large organizations. It is important to be able to tell in fine detail what was changed, when, and by whom.
+
+The Snowplow Console's UI offers excellent facilities to get started quickly with Data Structures (either using the Builder or directly the JSON editor), and is a solid tool for smaller teams; but it doesn't implement such approval processes, neither does it offer such fine-grained visibility around changes.
+
+A common solution when faced with these requirements is to move management to some form of version control platform (github/gitlab). This opens up an entire ecosystem of tools and patterns enabling all manner of custom workflows.
+
+We have built [snowplow-cli](/docs/understanding-tracking-design/managing-your-data-structures/cli/) to help you bridge the gap between these repository based workflows and BDP Console.
+
+## Prerequisites
+
+* A deployed Snowplow BDP pipeline
+* [snowplow-cli](/docs/understanding-tracking-design/managing-your-data-structures/cli/#download) downloaded and configured
+* A familiarity with [git](https://git-scm.com/) and an understanding of [github workflows](https://docs.github.com/en/actions/writing-workflows)
```

Comment:
I think the product is called `github actions`, but Im not sure what's the difference

Reply:
I thought that too but their docs are a bit confusing. They refer to workflows in a lot of places then actions in their more detailed stuff.

Reply:
Don't know, need some marketing people. I would use actions since it's a more recognisable name

Reply:
As far as I can tell, actions is the platform and workflows are the things you create. Yeah, I can change it to actions to avoid anyone who hasn't read the definitions getting confused.